ols/src/llms/llm_loader.py

```diff
@@ -54,44 +46,55 @@ class LLMLoader:
         params = {'temperature': 0.02, 'top_p': 0.95}
 
         bare_llm = LLMLoader(provider="openai", model="gpt-3.5-turbo", params=params)
-        llm_chain = LLMChain(llm=bare_llm, prompt=prompt)
+        llm_chain = LLMChain(llm=bare_llm.llm, prompt=prompt)
         ```
     """
 
     def __init__(
         self,
-        provider: Optional[str] = None,
-        model: Optional[str] = None,
+        provider: str,
+        model: str,
         params: Optional[dict] = None,
+        llm_providers: Optional[LLMProviders] = None,
     ) -> None:
         """Initialize LLM loader.
 
         Args:
             provider: The LLM provider name.
             model: The LLM model name.
             params: The parameters to override and pass to the LLM backend.
+            llm_providers: The LLM providers configuration.
```

Comment:
when would we expect a caller to pass this in, rather than llm_loader using the OLS config?